pkg/cloudprovider/controller/nodecontroller.go

```diff
@@ -509,11 +537,17 @@ func (nc NodeController) tryUpdateNodeStatus(node *api.Node) (error, time.Durati
 				nc.recordNodeOfflineEvent(node)
 			}
 		}
-		_, err = nc.kubeClient.Nodes().Update(node)
-		if err != nil {
-			glog.Errorf("Error updating node %s: %v", node.Name, err)
-		} else {
-			return nil, gracePeriod, lastReadyCondition
+		if !api.Semantic.DeepEqual(oldNodeStatus.status, node.Status) {
```

Comment:
Why did you put this in an if, where previously it was unconditional? Just as an optimization?